systers_portal/meetup/views.py

```diff
@@ -801,79 +774,50 @@ def post(self, request):
-#                 searched_meetups = Meetup.objects.filter(Q(description__contains=keyword) |
-#                                                          Q(title__contains=keyword), date=date,
-#                                                          description__contains=keyword)
-#             elif meetup_location and keyword:
-#                 # search by meetup_location & keyword
-#                 searched_meetups = Meetup.objects.filter(Q(description__contains=keyword) |
-#                                                          Q(title__contains=keyword),
-#                                                          meetup_location__name=meetup_location)
-#             elif date:
-#                 # search by date
-#                 searched_meetups = Meetup.objects.filter(date=date)
-#             elif meetup_location:
-#                 # search by meetup_location
-#                 searched_meetups = Meetup.objects.filter(meetup_location__name=meetup_location)
-#             elif keyword:
-#                 # search by keyword
-#                 searched_meetups = Meetup.objects.filter(Q(description__contains=keyword) |
-#                                                          Q(title__contains=keyword))
-#             else:
-#                 searched_meetups = Meetup.objects.filter(date__gte=datetime.date.today())
-#
-#             results = list()
-#             unit = ''
-#             for meetup in searched_meetups:
-#                 distance = ''
-#                 if selected_filter == 'distance':
-#                     # if user filters by distance
-#                     geolocator = Nominatim(timeout=6)
-#                     user_loc = (geolocator.geocode(location))
-#                     user_point = Point((float)(user_loc.raw['lon']), (float)(user_loc.raw['lat']))
-#                     meetup_loc = (geolocator.geocode(meetup.meetup_location.location))
-#                     meetup_point = Point((float)(meetup_loc.raw['lon']),
-#                                          (float)(meetup_loc.raw['lat']))
-#                     distance = (int)(user_point.distance(meetup_point)) * 100
-#                     results.sort(key=operator.itemgetter('distance'))
-#                     unit = 'kilometers from your location'
-#
-#                 results.append({'date': meetup.date,
-#                                 'meetup': meetup.title,
-#                                 'location': meetup.meetup_location.name,
-#                                 'location_slug': meetup.meetup_location.slug,
-#                                 'meetup_slug': meetup.slug,
-#                                 'distance': distance,
-#                                 'unit': unit})
-#
-#             results.sort(key=operator.itemgetter('date'))
-#             return JsonResponse({'search_results': results}, safe=False)
+
+class UpcomingMeetupsSearchView(ListView):
+    """Search Upcoming Meetups By  Keyword and Filter Date and Distance"""
```

Comment:
Minor Change: Correct the spacing.

Reply:
@sakshi1499 That is the part of the formatting IG

Reply:
Between By and Keyword in the Docstring

Reply:
Oh ok nice catch, didnt see that. Thank you :)